src/test/java/com/commercetools/payment/handler/CommercetoolsLookUpPaymentsControllerTest.java

```diff
@@ -0,0 +1,93 @@
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class)
+@AutoConfigureMockMvc
+public class CommercetoolsLookUpPaymentsControllerTest extends PaymentIntegrationTest {
+
+    @Autowired
+    private MockMvcAsync mockMvcAsync;
+
+    @Autowired
+    private TenantConfigFactory tenantConfigFactory;
+
+    private TenantConfig tenantConfig;
+    private SphereClient sphereClient;
+    private CtpFacade ctpFacade;
+
+    @Before
+    public void setUp() throws Exception {
+        tenantConfig = tenantConfigFactory.getTenantConfig(MAIN_TEST_TENANT_NAME)
+                .orElseThrow(IllegalStateException::new);
+
+        ctpFacade = new CtpFacadeFactory(tenantConfig).getCtpFacade();
+
+        sphereClient = tenantConfig.createSphereClient();
+    }
+
+    @Test
+    public void onPaypalPaymentNotFound_shouldReturn404() throws Exception {
+        mockMvcAsync.performAsync(get(format("/%s/commercetools/look-up/payments/%s/", MAIN_TEST_TENANT_NAME, "nonExistingId")))
+                .andDo(print())
+                .andExpect(status().isNotFound())
+                .andExpect(content().contentType(APPLICATION_JSON_UTF8));
+    }
+
+    @Test
+    public void onValidPaymentId_shouldReturnPaypalPayment() throws Exception {
+        final String ctpPaymentId = createCartAndPayment(sphereClient);
+        mockMvcAsync.performAsync(post(format("/%s/commercetools/create/payments/%s", MAIN_TEST_TENANT_NAME, ctpPaymentId)))
+                .andDo(print())
+                .andReturn();
+
+        Optional<Payment> paymentOpt = executeBlocking(ctpFacade.getPaymentService().getById(ctpPaymentId));
+
+        String paypalPaymentId = paymentOpt.get().getInterfaceId();
+        MvcResult result = mockMvcAsync.performAsync(get(format("/%s/commercetools/look-up/payments/%s/", MAIN_TEST_TENANT_NAME, paypalPaymentId)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(APPLICATION_JSON_UTF8))
+                .andReturn();
+
+        String responseAsString = result.getResponse().getContentAsString();
+        JsonNode responseAsJson = SphereJsonUtils.parse(responseAsString);
+        ObjectNode responseBodyAsJson = (ObjectNode) SphereJsonUtils.parse(responseAsJson.path("responseBody").asText());
```

Comment:
You don't need all these casts here: `MvcResult` already has built-in methods to verify JSON. See examples:
https://github.com/commercetools/commercetools-paypal-plus-integration/blob/79aaab4ac069fa92a973f01525db2e36a77da285/src/test/java/com/commercetools/payment/handler/CommercetoolsHealthControllerIT.java#L35-L36

https://github.com/commercetools/commercetools-paypal-plus-integration/blob/3fe8b33d91db2bdbfe9d9fcb7c3e374a4fb11d78/src/test/java/com/commercetools/payment/handler/CommercetoolsExecutePaymentsControllerIT.java#L100-L124

https://github.com/commercetools/commercetools-paypal-plus-integration/blob/3fe8b33d91db2bdbfe9d9fcb7c3e374a4fb11d78/src/test/java/com/commercetools/payment/handler/CommercetoolsCreatePaymentsControllerIT.java#L169-L193

Reply:
Thanks for the tip, but I'm not sure if this applies in this case. I put in already a String before sending back a response, not Java object.

Reply:
16a2575